app/controllers/search_controller.rb

```diff
@@ -7,41 +8,83 @@ def index
 
     @trips ||= []
     @search = Search.new(params[:search])
+    the_number_of_trips_per_page = 10
+    the_first_date = [(@search.date_value) -31, (Date.today) -31].min # we need trips of the past month to say "hey there was some trips last month" if no results
+    the_last_date =  (@search.date_value) +31
     if @search.valid?
       if @search.from_lon.present? && @search.to_lon.present?
-        @trips = Trip
+        the_trips_around_the_requested_day = Trip
                    .unscoped
                    .includes(:points)
-                   .where('departure_date >= ?', @search.date_value)
+                   .where('departure_date >= ?', the_first_date)
+                   .where('departure_date <= ?', the_last_date)
                    .where(state: 'confirmed')
                    .from_to(@search.from_lon, @search.from_lat, @search.to_lon, @search.to_lat)
                    .order(departure_date: :asc)
                    .order(departure_time: :asc)
-                   .page(params[:page]).per(10)
+                   #.page(params[:page]).per(the_number_of_trips_per_page)
       elsif @search.from_lon.present?
-        @trips = Trip
+        the_trips_around_the_requested_day = Trip
                    .unscoped
                    .includes(:points)
-                   .where('departure_date >= ?', @search.date_value)
+                   .where('departure_date >= ?', the_first_date)
+                   .where('departure_date <= ?', the_last_date)
                    .where(state: 'confirmed')
                    .from_only(@search.from_lon, @search.from_lat)
                    .order(departure_date: :asc)
                    .order(departure_time: :asc)
-                   .page(params[:page]).per(10)
+                   #.page(params[:page]).per(the_number_of_trips_per_page)
       elsif @search.to_lon.present?
-        @trips = Trip
+        the_trips_around_the_requested_day = Trip
                    .unscoped
                    .includes(:points)
-                   .where('departure_date >= ?', @search.date_value)
+                   .where('departure_date >= ?', the_first_date)
+                   .where('departure_date <= ?', the_last_date)
                    .where(state: 'confirmed')
                    .to_only(@search.to_lon, @search.to_lat)
                    .order(departure_date: :asc)
                    .order(departure_time: :asc)
-                   .page(params[:page]).per(10)
+                   #.page(params[:page]).per(the_number_of_trips_per_page)
       end
     end
+    # set .the_previous_trip__departure_date attribute
```

Comment:
Code below is too fat for error probe check and understanding
You can use private submethod or create a dedicated service / Search

Reply:
Poke me if you need help